5.sc_Morphology_Logistic_Regression/0.split_data/scripts/split_data.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import ast
+import itertools
+import pathlib
+
+import numpy as np
+import pandas as pd
+import pyarrow.parquet as pq
+import toml
+from joblib import dump
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV, StratifiedKFold, train_test_split
+from sklearn.utils import parallel_backend, shuffle
+
+# In[2]:
+
+
+# Parameters
+cell_type = "SHSY5Y"
+aggregation = False
+nomic = False
+flag = True
+control = "DMSO_0.100_DMSO_0.025"
+treatment = "Thapsigargin_1.000_DMSO_0.025"
+
+
+# In[3]:
+
+
+print(cell_type, aggregation, nomic, flag, control, treatment)
+
+
+# In[4]:
+
+
+if flag == False:
+    # read in toml file and get parameters
+    toml_path = pathlib.Path("../1.train_models/single_class_config.toml")
+    with open(toml_path, "r") as f:
+        config = toml.load(f)
+    f.close()
```

Comment:
I may be mistaken here but I think you don't need the `f.close()` if you open a file with `with open(toml_path, "r")`.

This is explained at https://realpython.com/why-close-file-python/

> At some point in your Python coding journey, you learn that you should use a context manager to open files. Python context managers make it easy to close your files once you’re done with them:
> 
> with open("hello.txt", mode="w") as file:
>     file.write("Hello, World!")
> The [with statement](https://realpython.com/python-with-statement/) initiates a context manager. In this example, the context manager [opens](https://realpython.com/read-write-files-python/#opening-and-closing-a-file-in-python) the file hello.txt and manages the file resource as long as the context is active. In general, all the code in the indented block depends on the file object being open. Once the indented block either ends or raises an exception, then the file will close.
> 
> If you’re not using a context manager or you’re working in a different language, then you might explicitly close files

Reply:
Good to know. I had thought that one had to close files after opening them. Thank you for teaching me something new!